src/main/kotlin/com/hibob/academy/employeeFeedback/service/FeedbackService.kt

```diff
@@ -0,0 +1,43 @@
+package com.hibob.academy.employeeFeedback.service
+
+import com.hibob.academy.employeeFeedback.dao.*
+import jakarta.ws.rs.BadRequestException
+import org.springframework.stereotype.Service
+
+@Service
+class FeedbackService(private val feedbackDao: FeedbackDao, private val employeeDao: EmployeeDao) {
+    fun submitFeedback(employeeId: Int, companyId: Int, anonymity: AnonymityType, feedback: String): Boolean {
+        var employeeIdForFeedback: Int? = employeeId
+        if (anonymity == AnonymityType.ANONYMOUS)
+            employeeIdForFeedback = null
+
+        if (feedback.length < 30 || feedback.length > 500)
+            return false
+
+        val feedbackForSubmission =
+            FeedbackForSubmission(employeeIdForFeedback, companyId, anonymity, feedback)
+        feedbackDao.submitFeedback(feedbackForSubmission)
+        return true
+    }
+
+    fun viewAllSubmittedFeedback(employeeId: Int, companyId: Int): List<FeedbackInfo> {
+        return feedbackDao.viewAllSubmittedFeedback(companyId)
+    }
+
+    fun viewStatusOfMyFeedback(employeeId: Int, companyId: Int, feedbackId: Int): Map<Int, Boolean> {
+        val feedbackInfo = feedbackDao.getFeedbackById(feedbackId, companyId)
+        feedbackInfo?.let{
+            val idOfEmployeeThatWroteFeedbackDao = feedbackInfo.employeeId
+                ?: throw Exception("Can't check status of anonymous feedback")
+            if (employeeId != idOfEmployeeThatWroteFeedbackDao)
+                throw BadRequestException("Unauthorized to view this feedback status")
```

Comment:
also, same comment here as before. These are bunch of validations, how will the code look like if you need to add another validation?

Think about how you structure your function.

Reply:
Also too many if's and excpetion throwing in. one block can be confusing when reading the code.
Think of other developers that need to maintain it later